### ISSUE TITLE
Transport exceptions from thread to main

### ISSUE THREAD

**Issue by jasonivey** [jasonivey/mmotd#58]
Threads are no longer leaking exceptions.  But now the errors are simply being ignored.  Add library or logic (i.e. `current_exception`, `make_exception_ptr` and `rethrow_exception`) to transport the exceptions being thrown in threads to be caught and handled in `main`.

**Comment by jasonivey** [comment]
I believe this is still somewhat of an issue.  But I also believe that I did much to stem the tide from abandoning threads which have thrown an exception and there was no code to catch it.

In the existing `boost::thread::pool` tasks are dispatched.  Once the task is complete, it returns to the pool to be allocated again by another task.  This leaves the hole in the design where I don't know if I should try and fully handle all exceptions within each task?  Or should I leave exception catching out of the tasks and leave it in the `boost::thread::pool`
